bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQFridgeHandler.java

```diff
@@ -0,0 +1,388 @@
+        FridgeCapability refCap;
+        try {
+            refCap = getCapabilities();
+        } catch (LGThinqApiException e) {
+            logger.error("Error getting capability of the device. It's mostly like a bug", e);
+            return 0;
+        }
+        // temperature channels are little different. First we need to get the tempUnit in the first snapshot,
+        final Map<String, String> convertionMap = getConvertionMap(ch, refCap);
+        final Map<String, String> invertedMap = new HashMap<>();
+        convertionMap.forEach((k, v) -> {
+            invertedMap.put(v, k);
+        });
+
+        String strValue = invertedMap.get(value.toString());
+        if (strValue == null) {
+            logger.error("Temperature value informed can't be converted based on the cap file. It mostly like a bug");
+            return 0;
+        }
+        try {
+            return Integer.valueOf(strValue);
+        } catch (Exception ex) {
+            logger.error("Temperature value converted can't be cast to Integer. It mostly like a bug", ex);
+            return 0;
+        }
+    }
+
+    private Map<String, String> getConvertionMap(ChannelUID ch, FridgeCapability refCap) {
+        Map<String, String> convertionMap;
+        if (fridgeTempChannelUID.equals(ch)) {
+            convertionMap = RE_TEMP_UNIT_FAHRENHEIT.equals(tempUnit) ? refCap.getFridgeTempFMap()
+                    : refCap.getFridgeTempCMap();
+        } else if (freezerTempChannelUID.equals(ch)) {
+            convertionMap = RE_TEMP_UNIT_FAHRENHEIT.equals(tempUnit) ? refCap.getFreezerTempFMap()
+                    : refCap.getFreezerTempCMap();
+        } else {
+            throw new IllegalStateException("Conversion Map Channel temperature not mapped. It's most likely a bug");
+        }
+        return convertionMap;
+    }
+
+    @Override
+    public LGThinQApiClientService<FridgeCapability, FridgeCanonicalSnapshot> getLgThinQAPIClientService() {
+        return lgThinqFridgeApiClientService;
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return logger;
+    }
+
+    protected DeviceTypes getDeviceType() {
+        return DeviceTypes.AIR_CONDITIONER;
+    }
+
+    @Override
+    public String getDeviceAlias() {
+        return emptyIfNull(getThing().getProperties().get(PROP_INFO_DEVICE_ALIAS));
+    }
+
+    @Override
+    public String getDeviceUriJsonConfig() {
+        return emptyIfNull(getThing().getProperties().get(PROP_INFO_MODEL_URL_INFO));
+    }
+
+    @Override
+    public void onDeviceRemoved() {
+    }
+
+    @Override
+    public void onDeviceDisconnected() {
+    }
+
+    @Override
+    public void updateChannelDynStateDescription() throws LGThinqApiException {
+        FridgeCapability cap = getCapabilities();
+        manageDynChannel(icePlusChannelUID, CHANNEL_FR_ICE_PLUS, "Switch", !cap.getIcePlusMap().isEmpty());
+        manageDynChannel(expressFreezeModeChannelUID, CHANNEL_FR_EXPRESS_FREEZE_MODE, "String",
```

Comment:
Applies to the others here too. Use constants when possbible
```suggestion
        manageDynChannel(expressFreezeModeChannelUID, CHANNEL_FR_EXPRESS_FREEZE_MODE, "String",
```

Reply:
Done